clientUtils/patchHelper.test.ts

```diff
@@ -0,0 +1,84 @@
+#! /usr/bin/env jest
+
+import { setValueRecursive } from "./patchHelper"
+
+interface PatchHelperTestCase {
+    pointer: string[]
+    newValue: any
+    jsonValueBefore: any
+    jsonValueAfter: any
+}
+
+const testCases: PatchHelperTestCase[] = [
+    {
+        pointer: ["a", "b"],
+        newValue: 2,
+        jsonValueBefore: {},
+        jsonValueAfter: { a: { b: 2 } },
+    },
+    {
+        pointer: ["a", "b", "c"],
+        newValue: 2,
+        jsonValueBefore: {},
+        jsonValueAfter: { a: { b: { c: 2 } } },
+    },
+    {
+        pointer: ["a", "0", "b"],
+        newValue: 2,
+        jsonValueBefore: {},
+        jsonValueAfter: { a: [{ b: 2 }] },
+    },
+    {
+        pointer: ["a", "b", "0"],
+        newValue: 2,
+        jsonValueBefore: {},
+        jsonValueAfter: { a: { b: [2] } },
```

Comment:
If the `pointer` was instead `["a", "b", "1"]`, what would the expected output be? Or is that not allowed?

Reply:
It would be the same expected value - anything beyond the length of the current array is just appended

Reply:
Let's add a test case for that?